benchmarks/python/test_cat.py

```diff
@@ -0,0 +1,495 @@
+        T136 = fd.ops.cat([T135, T116], dim=-1, manual_padding=0)
+        T142 = fd.ops.broadcast_in_dim(
+            T35, shape=[1, 4, 2048, 128], broadcast_dims=[0, 1, 2, 3]
+        )
+        T143 = fd.ops.cast(T136, dtype=DataType.Float)
+        T144 = fd.ops.cast(T142, dtype=DataType.Float)
+        T145 = fd.ops.mul(T143, T144)
+        T146 = fd.ops.add(T100, T145)
+        T147 = fd.ops.cast(T146, dtype=DataType.BFloat16)
+        T154 = fd.ops.broadcast_in_dim(
+            T147, shape=[1, 4, 1, 2048, 128], broadcast_dims=[0, 1, 3, 4]
+        )
+        T161 = fd.ops.broadcast_in_dim(
+            T154, shape=[1, 4, 7, 2048, 128], broadcast_dims=[0, 1, 2, 3, 4]
+        )
+        T167 = fd.ops.reshape(T161, new_shape=[1, 28, 2048, 128])
+        T174 = fd.ops.broadcast_in_dim(
+            T23, shape=[1, 4, 1, 2048, 128], broadcast_dims=[0, 1, 3, 4]
+        )
+        T181 = fd.ops.broadcast_in_dim(
+            T174, shape=[1, 4, 7, 2048, 128], broadcast_dims=[0, 1, 2, 3, 4]
+        )
+        T187 = fd.ops.reshape(T181, new_shape=[1, 28, 2048, 128])
+        T188 = fd.ops.stride_order(T91, stride_order=[3, 2, 1, 0])
+        T189 = fd.ops.stride_order(T167, stride_order=[3, 2, 1, 0])
+        T190 = fd.ops.stride_order(T187, stride_order=[3, 2, 1, 0])
+        fd.add_output(T23)
+        fd.add_output(T147)
+        fd.add_output(T188)
+        fd.add_output(T189)
+        fd.add_output(T190)
+
+    inputs = [
+        torch.randn(
+            size=(2048, 512),
+            dtype=torch.bfloat16,
+            layout=torch.strided,
+            device="cuda",
+            requires_grad=False,
+        ),
+        torch.randn(
+            size=(1, 2048, 512),
+            dtype=torch.bfloat16,
+            layout=torch.strided,
+            device="cuda",
+            requires_grad=False,
+        ),
+        torch.randn(
+            size=(1, 2048, 128),
+            dtype=torch.bfloat16,
+            layout=torch.strided,
+            device="cuda",
+            requires_grad=False,
+        ),
+        torch.randn(
+            size=(1, 2048, 128),
+            dtype=torch.bfloat16,
+            layout=torch.strided,
+            device="cuda",
+            requires_grad=False,
+        ),
+        torch.randn(
+            size=(1, 28, 2048, 128),
+            dtype=torch.bfloat16,
+            layout=torch.strided,
+            device="cuda",
+            requires_grad=False,
+        ),
+        torch.randn(
+            size=(1, 4, 2048, 128),
+            dtype=torch.bfloat16,
+            layout=torch.strided,
+            device="cuda",
+            requires_grad=False,
+        ),
+    ]
+
+    if executor == "torchcompile":
+
+        def benchmark_fn(inputs):
+            return to_be_compiled(*inputs)
+
+        benchmark_fn = with_executor(executor, benchmark_fn)
+
+    elif executor == "thunder":
+        with FusionDefinition() as fd:
+            nvfuser_fusion_id4(fd)
+
+        inputs = [
```

Comment:
Why did you need to define the inputs separately?

Reply:
It exploded without that; thunder must be transforming things before it gets to nvfuser and that gets lost along the two paths.

anyway this is basically equivalent to Jie's question below so i'm going to resolve this and leave the next one open so we just track it in 1 place.